.github/workflows/ci.yml

```diff
@@ -49,26 +50,24 @@ jobs:
         steps:
         -   uses: actions/checkout@v4
 
-        -   name: Cache Python dependencies
-            uses: actions/cache@v4
-            with:
-                path: ~/.cache/pip
-                key: pip-${{ matrix.python-version }}-tests-${{ hashFiles('**/setup.json') }}
-                restore-keys:
-                    pip-${{ matrix.python-version }}-tests
-
         -   name: Set up Python ${{ matrix.python-version }}
             uses: actions/setup-python@v5
             with:
                 python-version: ${{ matrix.python-version }}
 
+        - name: Setup uv
+          uses: astral-sh/setup-uv@v6
+          with:
+              activate-environment: true
+              python-version: '3.9'
+
         -   name: Install Python dependencies
-            run: pip install -e .[tests]
+            run: uv pip install -e .[tests]
 
         -   name: Run pytest
             env:
                 AIIDA_WARN_v3: 1
-            run: pytest -sv tests
+            run: pytest tests
```

Comment:
The verbose output is excessive and it's hard to find the actual errors. We made a similar change in aiida-core recently.